core/src/test/java/apoc/spatial/GeocodeTest.java

```diff
@@ -34,8 +45,95 @@ public void testWrongUrlButViaOtherProvider() throws Exception {
         testGeocodeWithThrottling("osm", false, 
                 map("url", "https://api.opencagedata.com/geocode/v1/json?q=PLACE&key=KEY111"));
     }
+
+    @Test
+    public void testGeocodeWithBlockedAddressWithApocConf() {
+        final String geocodeBaseConfig = Geocode.PREFIX + "." + "opencage";
+        apocConfig().setProperty(geocodeBaseConfig + ".key", "myKey");
+        
+        Stream.of("https", "http", "ftp").forEach(protocol -> {
+            final String nonBlockedUrl = String.format(URL_FORMAT, protocol, NON_BLOCKED_ADDRESS);
+            final String nonBlockedReverseUrl = String.format(REVERSE_URL_FORMAT, protocol, NON_BLOCKED_ADDRESS);
+            
+            final String geocodeConfigUrl = geocodeBaseConfig + ".url";
+            final String geocodeConfigReverseUrl = geocodeBaseConfig + ".reverse.url";
+            
+            // check that if either url or reverse address are blocked 
+            // respectively the apoc.spatial.geocode and the apoc.spatial.reverseGeocode procedure fails
+            apocConfig().setProperty(geocodeConfigUrl, nonBlockedUrl);
+            apocConfig().setProperty(geocodeConfigReverseUrl, String.format(REVERSE_URL_FORMAT, protocol, BLOCKED_ADDRESS));
+
+            assertGeocodeFails(true, BLOCKED_ERROR);
+            
+            apocConfig().setProperty(geocodeConfigUrl, String.format(URL_FORMAT, protocol, BLOCKED_ADDRESS));
+            apocConfig().setProperty(geocodeConfigReverseUrl, nonBlockedReverseUrl);
+            
+            assertGeocodeFails(false, BLOCKED_ERROR);
+
+            // check that if neither url nor reverse url are blocked 
+            // the procedures continue the execution (in this case by throwing a `java.net` tException)
+            apocConfig().setProperty(geocodeConfigUrl, nonBlockedUrl);
+            apocConfig().setProperty(geocodeConfigReverseUrl, nonBlockedReverseUrl);
+            
+            assertGeocodeFails(false, JAVA_NET_EXCEPTION);
+            assertGeocodeFails(true, JAVA_NET_EXCEPTION);
```

Comment:
Oh ok, I see what's happening. It fails but because of a different problem, which means that IP blocking worked. Perhaps this is a little awkward to understand, do you think there is a way we could make the query succeed when the IP is not blocked?

Reply:
Unfortunately I don't think so, it would take an opencage api key to make that query succeed.
However I changed the name of these methods from `assertGeocodeFails ` to `assertGeocodeAllowedUrl`, at least to be a bit clearer.